### ISSUE TITLE
Be selective in deletion of non-generated guards

### ISSUE THREAD

**Issue by wiegell** [rhys-vdw/ts-auto-guard#121]
It's currently not feasible to do a simple search of all .ts files and hand them over to ts-auto-guard like this 
`find ./src -type f -name "*.ts" | yarn ts-auto-guard` since all manually generated .guard.ts files will be deleted.
A suggestion could be to only delete guards that has the autogenerated header:
```
/*
 * Generated type guards for "edit-diag.model.ts".
 * WARNING: Do not manually change this file.
 */
```
Or as an alternative only delete .guard.ts files, where a corresponding file with `@see` is actually present.

**Comment by rhys-vdw** [comment]
Hey @wiegell, thanks for the report.

> It's currently not feasible to do a simple search of all .ts files and hand them over to ts-auto-guard like this

If you don't include paths then it'll automatically scan all files in your project. Should this be clearer in the docs?

However, if you _do_ want to just regenerate some subset of files, then you will hit the problem you've described.

The only reason it deletes guard files at all (as opposed to just rewriting them) is to clean up old files whose corresponding types or `@see` comments have been removed.

I think a cheap fix would be to add a flag `--skip-clean` that simply skips over the deletion [here](https://github.com/rhys-vdw/ts-auto-guard/blob/a749c59f09fdd00c49db96dd235225fe01f02a59/src/index.ts#L731-L733), and then lets the generated files [overwrite those that already exist(https://github.com/rhys-vdw/ts-auto-guard/blob/a749c59f09fdd00c49db96dd235225fe01f02a59/src/index.ts#L790-L794).

Just a thought on this, but I think maybe there'd be issues regenerating a guard that references a type that is not also included... tbh I added the paths option early on and haven't messed with it much.

Anyway, you're more than welcome to open a PR to tackle this one. Happy for features to be guided by active users.

**Comment by wiegell** [comment]
Thank you for your response.

When i just run ts-auto-guard from project root it also deletes all manually created guards.
Tbh. i think it should be the other way around - no deletion unless a `--clean` flag is provided. At least when it's run with no path provided.
I'm sorry, but i don't have time atm to contribute myself.

**Comment by rhys-vdw** [comment]
@wiegell yep, skipping 'clean' by default seems fine to me! I won't be adding this features but happy to accept a PR for it.
